### PR TITLE
Help Center: implement session continuity

### PR DESCRIPTION
**Pull request by alshakero**
💡 **Make sure to review with `whitespace` ignored (add `?w=1` to the diff URL).**

#### Proposed Changes

* This makes the Happychat sessions persistent across navigation and page refresh in the help-center. It does that by checking if the user has an active Happychat session, and if they do, it re-opens the help-center and redirects it to the chat window.

I'm aware the diff is far too large, but I promise most files have 1-4 lines of change.

#### How does it work
The principle is pretty simple.
1. Store Help Center's open/closed state in the help-center data store, and remove this state from Calypso and anywhere else.
2. On page load, check if the user has any active Happychat session. If they do, reopen the help center.

#### Testing Instructions

Set config entry `calypso/help-center` to `true` in `config/development.json`.

#### Setting the environment
1. cd into `apps/happychat`.
2. run `yarn dev --sync`.
3. sandbox widgets.wp.com.
4. start Calypso `yarn start` (in root folder).

#### Live chat
1. Login at https://hud-staging.happychat.io/ to accept your own chat.
2. Go to http://calypso.localhost:3000/pages.
3. Start a chat in the help-center.
4. Refresh the page.
5. The chat session should continue as is without sending the intro message again.
6. As the chat operator, type `/end` to end the chat.
7. Refresh the page.
8. The help-center shouldn't re-open.

#### From Calypso to the editor
1. Login at https://hud-staging.happychat.io/ to accept your own chat.
2. Go to http://calypso.localhost:3000/pages.
3. Start a chat in the help-center.
4. Refresh the page.
5. The chat session should continue as is without sending the intro message again.
6. As the chat operator, type `/end` to end the chat.
7. Refresh the page.
8. The help-center shouldn't re-open.

### CODE REVIEW

File: packages/help-center/src/components/help-center.tsx
Comment (alshakero) [inline packages/help-center/src/components/help-center.tsx]: I removed the loading blockage because it loads for nothing when the help center lands in the chat after a navigation. I moved the loading blockage to the "Still need help?" button.

File: packages/help-center/src/components/history-recorder.tsx
Comment (alshakero) [inline packages/help-center/src/components/history-recorder.tsx]: This component records every navigation in the help-center store, the help-center store then persist that.

File: packages/help-center/src/utils.ts
Comment (alshakero) [inline packages/help-center/src/utils.ts]: This is just to make testing easier.

Comment (klimeryk) [reply packages/help-center/src/utils.ts]: Or it's a sneaky way to launch it to 100% users :trollface:

File: packages/data-stores/src/help-center/reducer.ts
Comment (sirbrillig) [inline packages/data-stores/src/help-center/reducer.ts]: Really minor:

```suggestion
import type { Location } from './types';
```

**Comment by escapemanuele** [comment]
The job you did is amazing 🚀 

I noticed that articles are maintained only if they are contextual, not searched, one tho :)

https://user-images.githubusercontent.com/52076348/183855256-422523af-da98-4cde-94dc-b5d5e5539aab.mp4

**Comment by renancarvalho** [comment]
Great Job Omar,General remarks:

**Comment by renancarvalho** [comment]
Great job @alshakero, I have some general remarks I have found while testing.

** Checked on simple and atomic site:**

- When clicking on an article while testing `Opening an article and Support options`
 It works, I just noticed some errors on `devtools`, I am not sure if they are related to these changes, but good to double check. 
image

- When searching for an article, for instance using the word 'plans' and then clicking on an article item. I believe it is saving the search criteria on localStorage(?), then when clicking on the 'X' on the `searchbox` and refreshing the page is not removing it from the place where it is saved. Please see the recording for better understanding.

https://user-images.githubusercontent.com/2653810/183862669-4d46e7b8-0e11-488b-bd2d-b4c3416514a6.mp4

- Live chat, Maybe I am missing a step of the testing instructions, but I am getting this.

image

and some errors on devTools 

image


- Calypso Iframe Bridge, I did see the HC icon on the top, but the page itself did not load.
image


Let me know if you have any questions.

**Comment by alshakero** [comment]
Thanks a lot @escapemanuele and @renancarvalho! I addressed everything.

**Comment by alshakero** [comment]
@renancarvalho regarding Happychat and The bridge. It seems you're proxied and your sandbox isn't up-to-date. Either disable sandboxing or `git pull` in your sandbox.

**Comment by heavyweight** [comment]
Screenshot 2022-08-10 at 14 16 24
Once I open the chat the ui is blocked and can't type anymore. Probably related to the error

File: packages/help-center/src/components/help-center-contact-page.tsx
Comment (klimeryk) [inline packages/help-center/src/components/help-center-contact-page.tsx:132]: Looks like this part is not working as expected:
Screenshot 2022-08-11 at 14 37 37

The link _is_ working, but seems `to` does not accept an `Object` (or maybe not in this shape?).

**Review by klimeryk — CHANGES_REQUESTED** [review]
To preface: I've tested only in Calypso (Themes, Plans, Add-ons pages usually).

> This makes the help-center sessions persistent across navigation and page refresh. 

Could you clarify if this only applies to Happy Chat state or the whole Help Center?
I was ready to submit a ticket and then refreshed. On refresh, only the ticket body was persisted, but not the subject:

https://user-images.githubusercontent.com/3392497/184155511-73c83bf7-db1b-49f0-8b12-171bb2dfb2e7.mov

**Blocker**: Looks like there's a fundamental change how chatting workflow works. With the FAB, the user could close the chat window, but they'd still be able to get back to chat via a small prompt that replaced the FAB while the chat was active to ensure they don't start new tickets/chats:

https://user-images.githubusercontent.com/3392497/184163218-98ec0a36-9b2d-498b-8120-61360eb2bf57.mov

**Blocker**: However, when the Help Center is closed, re-opening it seems to give you only option to start a new chat (or ticket) again. Interestingly, it seems the remnants of FAB _are_ active and suddenly that prompt I mentioned above is also visible...

https://user-images.githubusercontent.com/3392497/184163935-de9acfe4-3517-48df-9638-0b2b9254a000.mov

The FAB should not load at all, of course. But also, the Help Center should be able to detect there's an active session and load it (continue it). What's worse, that original chat is still active for the HE and they might continue replying to the user there. But none of that is shown to the user. What was the expected workflow here?

Super nitpick, but seems the tooltip gets hidden behind part of the Help Center window - but only when it's in its collapsed state:
Screenshot 2022-08-11 at 14 23 27

**High priority** Actually, looks like a bigger issue: when there's an active chat and the Help Center is minimized, the `Help Center` label is not visible, it's just a white rectangle:
Screenshot 2022-08-11 at 15 00 48

**High priority**: When there's a new message, there's no clear indication. The FAB would turn to ping color at least. But it seem the Help Center... shows the cut off minimize tooltip? No, seriously, I could reproduce it reliably:
Screenshot 2022-08-11 at 14 59 51
I'm guessing it's some weird DOM/HTML quirk, but end result is that the user might miss they have a new message.

Related to this we _need_ to allow users to have the same chat open in more than one tab. That's easily doable with FAB, but I cannot figure out how to do it with the Help Center. Seems only the tab that opened the chat can show that chat. Users often have many tabs, will switch between them. If we only show active chat in one of them, they will end up creating new sessions instead.

**High priority**: Another change of behaviour: when the chat is ended, the Help Center immediately reverts to the main view. That's a very sudden change. From HEs' experience, users often come back with "one more question!" or "yes, I'm still here!" and that's something we need to support. They might also want to keep the chat around for a bit longer to read again what the HE shared, etc. AFAIK, the idea is that there's no clear "end of chat". It's one continuous one and the user decides when they close the window. Can we think how we could handle that more gracefully?

I noticed the situation with scrollbars is a bit out of hand and looks very messy:
Screenshot 2022-08-11 at 15 04 31
Why is there a space between the innermost scrollbar and the Help Center one? Seems Help Center doesn't need it at that size anyway.

Seems there's a bug in how the support docs suggestions under the contact forms are handled. If I'm on the Plans page, seems the `Explore the WordPress.com Plans` page is always on top. That's probably a good call (though let's make sure we follow-up with data). But it seems that when Sibyl tries to add its suggestions, that same page keeps getting re-added. So you end up with duplication or worse:

https://user-images.githubusercontent.com/3392497/184158341-d0a6e9e2-f479-4a63-bf7c-6b64ba4e3500.mov

Nitpick: the `What's new` link has an icon that usually means an external resource (as it does for all the other resources in this section), but it actually just opens a modal in the same window, which seems a bit unexpected:
Screenshot 2022-08-11 at 14 39 21

The `Recommended resources` section links open in new tab instead of inline (as the ones you search for). Is that expected? Also, it seems we don't track clicks to them (at least in the default view)? Could we add that? 🙏 

As mentioned on the report of the original issue, for whatever reason, the `howCanWeHelp` custom field is filled with the first message of the chat (instead of empty value or something else). Seems `language` is also set to `__UNKNOWN__`. It's important we have these values to have proper metrics and data to analyze 🙏

I tested with a free user and noticed there's a misaligned `?` icon on the forum form:

Screenshot 2022-08-11 at 15 19 46

It does not seem to display anything on hover (or click), either 🤔 

Unrelated to this PR, but I want to highlight again that this very bad UX:
Screenshot 2022-08-11 at 14 26 14

The CTA button is hidden below the fold - it can be very disorienting for users. Could we make it so that it's sticky to the bottom of the window and always visible? Should be very easy using modern CSS ([`position: sticky`](https://css-tricks.com/position-sticky-2/) maybe?) 🙏

**Comment by alshakero** [comment]
> Could you clarify if this only applies to Happy Chat state or the whole Help Center?
I was ready to submit a ticket and then refreshed. On refresh, only the ticket body was persisted, but not the subject:

The whole help center, but the `subject` field, is missing from the [persistence list](https://github.com/Automattic/wp-calypso/blob/fix/help-center-continuity/packages/data-stores/src/help-center/index.ts#L23-L29). 

> Blocker: Looks like there's a fundamental change how chatting workflow works. With the FAB, the user could close the chat window, but they'd still be able to get back to chat via a small prompt that replaced the FAB while the chat was active to ensure they don't start new tickets/chats:

Do you have any ideas on how to deal with this in the help center, should we disable the close button as long as a session is active and only keep the minimize one? WDYT?

> Blocker: However, when the Help Center is closed, re-opening it seems to give you only option to start a new chat (or ticket) again. 

Yes, this is by design, maybe we shouldn't allow closing as long as the session is active. 

> Interestingly, it seems the remnants of FAB are active and suddenly that prompt I mentioned above is also visible...
Wow, we didn't know about this, we disable the FAB when there is help center, but this seems to be another thing, we'll look into it.

> The FAB should not load at all, of course. But also, the Help Center should be able to detect there's an active session and load it (continue it). What's worse, that original chat is still active for the HE and they might continue replying to the user there. But none of that is shown to the user. What was the expected workflow here?

We thought navigating away from the chat and closing the socket connection is enough to indicate to the HE that the chat is at least interrupted. I looked and didn't find a way to signal to the HE that the user closed the chat. I only found that the operator could end it, so we handle this operator's message and close happychat when that arrives. But couldn't figure a away to end the session from the user's side. 

The expected workflow is like this:
-> The user starts a chat, they have the chat.
-> The user may navigate or refresh, they'll come back to the chat.
-> The user will have their problem solved.
-> The user may close the window and move on, or the HE can end the chat. 

They should always land in the chat unless they close the help center or the HE ends it.

We're obviously happy to change this.

> Super nitpick, but seems the tooltip gets hidden behind part of the Help Center window - but only when it's in its collapsed state:

Great catch. Will look into it.

> High priority Actually, looks like a bigger issue: when there's an active chat and the Help Center is minimized, the Help Center label is not visible, it's just a white rectangle:

> High priority: When there's a new message, there's no clear indication. The FAB would turn to ping color at least. But it seem the Help Center... shows the cut off minimize tooltip? No, seriously, I could reproduce it reliably:

This is certainly a new bug. It should say Help Center with the unread count next to it. Looking into it.

> Related to this we need to allow users to have the same chat open in more than one tab. That's easily doable with FAB, but I cannot figure out how to do it with the Help Center. Seems only the tab that opened the chat can show that chat. Users often have many tabs, will switch between them. If we only show active chat in one of them, they will end up creating new sessions instead.

It seems, detecting if there is an active chat session in the help center will solve an array of problems. If we check that and the answer is yes, we may open the help-center and land the user in Happychat without any further persistence across the help-center, that might simplify things a bit.

> High priority: Another change of behaviour: when the chat is ended, the Help Center immediately reverts to the main view. That's a very sudden change. From HEs' experience, users often come back with "one more question!" or "yes, I'm still here!" and that's something we need to support. They might also want to keep the chat around for a bit longer to read again what the HE shared, etc. AFAIK, the idea is that there's no clear "end of chat". It's one continuous one and the user decides when they close the window. Can we think how we could handle that more gracefully?

Yes. We'll remove the redirection on chat end. No problem.

**Comment by alshakero** [comment]
> Could you clarify if this only applies to Happy Chat state or the whole Help Center?
I was ready to submit a ticket and then refreshed. On refresh, only the ticket body was persisted, but not the subject:

The whole help center, but the `subject` field, is missing from the [persistence list](https://github.com/Automattic/wp-calypso/blob/fix/help-center-continuity/packages/data-stores/src/help-center/index.ts#L23-L29). 

> Blocker: Looks like there's a fundamental change how chatting workflow works. With the FAB, the user could close the chat window, but they'd still be able to get back to chat via a small prompt that replaced the FAB while the chat was active to ensure they don't start new tickets/chats:

Do you have any ideas on how to deal with this in the help center, should we disable the close button as long as a session is active and only keep the minimize one? WDYT?

> Blocker: However, when the Help Center is closed, re-opening it seems to give you only option to start a new chat (or ticket) again. 

Yes, this is by design, maybe we shouldn't allow closing as long as the session is active? WDYT?

> Interestingly, it seems the remnants of FAB are active and suddenly that prompt I mentioned above is also visible...
Wow, we didn't know about this, we disable the FAB when there is help center, but this seems to be another thing, we'll look into it.

> The FAB should not load at all, of course. But also, the Help Center should be able to detect there's an active session and load it (continue it). What's worse, that original chat is still active for the HE and they might continue replying to the user there. But none of that is shown to the user. What was the expected workflow here?

We thought navigating away from the chat and closing the socket connection is enough to indicate to the HE that the chat is at least interrupted. I looked and didn't find a way to signal to the HE that the user closed the chat. I only found that the operator could end it, so we handle this operator's message and close happychat when that arrives. But couldn't figure a away to end the session from the user's side. 

The expected workflow is like this:
-> The user starts a chat, they have the chat.
-> The user may navigate or refresh, they'll come back to the chat.
-> The user will have their problem solved.
-> The user may close the window and move on, or the HE can end the chat. 

They should always land in the chat unless they close the help center or the HE ends it.

We're obviously happy to change this.

> Super nitpick, but seems the tooltip gets hidden behind part of the Help Center window - but only when it's in its collapsed state:

Great catch. Will look into it.

> High priority Actually, looks like a bigger issue: when there's an active chat and the Help Center is minimized, the Help Center label is not visible, it's just a white rectangle:

> High priority: When there's a new message, there's no clear indication. The FAB would turn to ping color at least. But it seem the Help Center... shows the cut off minimize tooltip? No, seriously, I could reproduce it reliably:

This is certainly a new bug. It should say Help Center with the unread count next to it. Looking into it.

> Related to this we need to allow users to have the same chat open in more than one tab. That's easily doable with FAB, but I cannot figure out how to do it with the Help Center. Seems only the tab that opened the chat can show that chat. Users often have many tabs, will switch between them. If we only show active chat in one of them, they will end up creating new sessions instead.

It seems, detecting if there is an active chat session in the help center will solve an array of problems. If we check that and the answer is yes, we may open the help-center and land the user in Happychat without any further persistence across the help-center, that might simplify things a bit.

> High priority: Another change of behaviour: when the chat is ended, the Help Center immediately reverts to the main view. That's a very sudden change. From HEs' experience, users often come back with "one more question!" or "yes, I'm still here!" and that's something we need to support. They might also want to keep the chat around for a bit longer to read again what the HE shared, etc. AFAIK, the idea is that there's no clear "end of chat". It's one continuous one and the user decides when they close the window. Can we think how we could handle that more gracefully?

Yes. We'll remove the redirection on chat end. No problem.

> I noticed the situation with scrollbars is a bit out of hand and looks very messy. Why is there a space between the innermost scrollbar and the Help Center one? Seems Help Center doesn't need it at that size anyway.

Looking into it.

> Seems there's a bug in how the support docs suggestions under the contact forms are handled. If I'm on the Plans page, seems the Explore the WordPress.com Plans page is always on top. That's probably a good call (though let's make sure we follow-up with data). But it seems that when Sibyl tries to add its suggestions, that same page keeps getting re-added. So you end up with duplication or worse.

This is a React key issue probably will check it.

> Nitpick: the What's new link has an icon that usually means an external resource (as it does for all the other resources in this section), but it actually just opens a modal in the same window, which seems a bit unexpected:

Great point.

> The Recommended resources section links open in new tab instead of inline (as the ones you search for). Is that expected? Also, it seems we don't track clicks to them (at least in the default view)? Could we add that? 🙏

Yes, Sibyl articles are not inlined, maybe they need an external link icon?

> As mentioned on the report of the original issue, for whatever reason, the howCanWeHelp custom field is filled with the first message of the chat (instead of empty value or something else). Seems language is also set to __UNKNOWN__. It's important we have these values to have proper metrics and data to analyze 🙏

Regarding howCanWeHelp, we'll end an empy value since we don't ask that. And for language we'll send the user language, no problem.

> I tested with a free user and noticed there's a misaligned ? icon on the forum form:

Yup, Calypso CSS is messing with us. This one should have a popover explaining it.

> The CTA button is hidden below the fold - it can be very disorienting for users. Could we make it so that it's sticky to the bottom of the window and always visible? Should be very easy using modern CSS ([position: sticky](https://css-tricks.com/position-sticky-2/) maybe?) 🙏

Will check this.

**Comment by klimeryk** [comment]
Appreciate the thorough reply!

> The whole help center, but the subject field, is missing from the [persistence list](https://github.com/Automattic/wp-calypso/blob/fix/help-center-continuity/packages/data-stores/src/help-center/index.ts#L23-L29).

Hahaha, leave it to me to find it ;) Thanks for the link, glad to see it's just this one field.

> Do you have any ideas on how to deal with this in the help center, should we disable the close button as long as a session is active and only keep the minimize one? WDYT?

> Yes, this is by design, maybe we shouldn't allow closing as long as the session is active? WDYT?

The current flow leads to users "losing" the current chats (that HEs are actively connected to), without even ending them. That's not acceptable and I'm not sure how this was approved or designed. As long as the session has not been ended by the Happiness Engineer, the user needs to be able to easily go back to it. And have a clear indication on refreshes that the session is still active. For example, they might have a network issue or some other issue that requires them to refresh. They might close the tab and open a new one (maybe a link the HE sent). The session has to stay active and available in all windows. **This is a blocker** if it's not possible to implement in the current scope. We cannot ship this as-is, because it will lead to duplicate chats being open and confusion for both users and HEs.
There was a designer attached to this project that I'd assume has the whole context and requirements. Could we ask them to double-check how this flow is supposed to work according to their designs?

> -> The user may close the window and move on, or the HE can end the chat.

> They should always land in the chat unless they close the help center or the HE ends it.

Only the HE can close (`end`) the chat. And they _always_ do. So you don't have to worry about that part. Closing the chat window from user perspective should _not_ mean "I'm done with the chat". More like "I'm minimizing it". It's up to the HE to `end` the chat or not at that (or any) point.
The user _can_ keep the chat window open and keep on typing after the chat was supposedly `ended` by the HE. That's a feature, not a bug ;) That's part of how we handle support and we need to preserve it. Hope this makes it clearer!

> It seems, detecting if there is an active chat session in the help center will solve an array of problems. If we check that and the answer is yes, we may open the help-center and land the user in Happychat without any further persistence across the help-center, that might simplify things a bit.

Yup! AFAIK, there should already be an endpoint or message sent through the websocket about an active/open session. There's some surprising or confusing ways that Happy Chat handles sessions (they can linger for a while, user can re-open a closed one...), so I'm happy to sync-up on this more (maybe on a call) if it'd be helpful.

>  The Recommended resources section links open in new tab instead of inline (as the ones you search for). Is that expected? Also, it seems we don't track clicks to them (at least in the default view)? Could we add that? 🙏

> Yes, Sibyl articles are not inlined, maybe they need an external link icon?

Sorry, looks like I did not include screenshot for that. From what I remember those were the default support docs shown based on the current view/page - not something from Sibyl, as far as I know. But if they are indeed coming from Sibyl, then yup, showing an external link icon would help 💖

**Comment by kriskarkoski** [comment]
> Only the HE can close (end) the chat. And they always do. So you don't have to worry about that part. Closing the chat window from user perspective should not mean "I'm done with the chat". More like "I'm minimizing it". It's up to the HE to end the chat or not at that (or any) point.

It has been awhile since I've chatted, but I thought if the user left WordPress.com altogether (at least where Happychat loads) the session was ended after a minute or so. There's an example of it on 1565-gh-Automattic/happychat, I believe that message is shown pretty instantly when the user leaves, then it's actually ended after another minute or so.

File: packages/happychat-connection/src/use-happychat-available.ts
Comment (alshakero) [inline packages/happychat-connection/src/use-happychat-available.ts:34]: I changed this into a promise to be able to use `useQuery` caching instead of building our own.

File: packages/help-center/src/components/help-center-header.tsx
Comment (alshakero) [inline packages/help-center/src/components/help-center-header.tsx:56]: I kept this because it will be needed later for sure.

File: packages/help-center/src/components/help-center.tsx
Comment (alshakero) [inline packages/help-center/src/components/help-center.tsx]: This is the key change.

**Comment by alshakero** [comment]
Hi @klimeryk! This is ready to test from a continuity standpoint. The PR is already huge, if you like, please feel free to take it for a spin (already updated the testings steps). We'll address all the other issues (continuity) in follow-up PRs.

**Comment by alshakero** [comment]
🔴  Note: Don't merge before [reverting this line](https://github.com/Automattic/wp-calypso/pull/66407/files#diff-39269d9d1803b1fbf8cce528c6e6378f49b919cf345535129a7c99b6c47fd804R28).

**Comment by alshakero** [comment]
Hi @klimeryk! Any update on this? 🙏🏼

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/src/config.js
Comment (arcangelini) [inline apps/editing-toolkit/editing-toolkit-plugin/help-center/src/config.js:7]: Since we are getting ready to officially merge, just pointing out these things that we may want to change back.

File: packages/help-center/src/utils.ts
Comment (arcangelini) [inline packages/help-center/src/utils.ts]: Also need to remove this.

**Review by arcangelini — APPROVED** [review]
Everything looks good and tests well for me. However there seems to be something causing the chat to resend the opener message/data when I am not looking at the screen as a user.
Markup 2022-09-06 at 10 50 09

Markup 2022-09-06 at 10 49 59

I am approving this PR because this is something that I have addressed in the [upcoming PR](https://github.com/Automattic/wp-calypso/pull/67353) to address communications between HappyChat and HelpCenter window. If we plan on releasing this as well to users with this merge then maybe we would want to fix it in this PR.

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/src/config.js
Comment (alshakero) [reply apps/editing-toolkit/editing-toolkit-plugin/help-center/src/config.js:7]: Let's leave it like this in this PR, I'll revert this https://github.com/Automattic/wp-calypso/pull/66360 to have the right env at the right time.

File: client/layout/index.jsx
Comment (manzoorwanijk) [inline client/layout/index.jsx:82]: This causes a warning in the console `Undefined breakpoint used in `mobile-first-breakpoint` >760px`

Screenshot 2022-09-16 at 10 19 29 AM

CC: @Automattic/team-calypso

File: packages/happychat-connection/src/use-happychat-auth.ts
Comment (jsnajdr) [inline packages/happychat-connection/src/use-happychat-auth.ts:47]: Hi @alshakero :wave: I noticed this `Date.now()` key when reviewing the React Query upgrade in #74381 and I was wondering what's the purpose of this unique key. Every time this module is loaded, either on Calypso boot or when its async chunk is loaded, `key` will be initialized to a new unique value. What was wrong when the key was just `getHappychatAuth`?

The only difference I can see is persistence of the the query results to local storage. If you don't want to use the old persisted data after a Calypso reload, the unique `key` achieves that. But there's a much better way to do that, the `meta: { persist: false }` option that you can pass to `useQuery`.

Comment (alshakero) [reply packages/happychat-connection/src/use-happychat-auth.ts:47]: Hey! Yes, that was a bad hack on my part. I was completely baffled by the persistence because when I checked the RQ docs, persistence was off by default, but the response was being persisted, and it drove me mad. 

Especially the issue was intermittent (because this code in runs in the Gutenberg editor as well and there is no default persistence there) and I couldn't explain it, so I added the key. Only a month later or so, I learned we had a Calypso-level config that had persistence on and the `Date.now()` key was spamming the cache.

I just forgot to reverse this hack. I'll fix today.

Comment (jsnajdr) [reply packages/happychat-connection/src/use-happychat-auth.ts:47]: > when I checked the RQ docs, persistence was off by default

Yes, the `react-query` library doesn't do any persistence by default, the cache is only in memory. But in Calypso we're setting it up with the persistence plugin, and it's on by default for all queries, you need to opt-out manually with the `meta.persist: false` flag.

Comment (heavyweight) [reply packages/happychat-connection/src/use-happychat-auth.ts:47]: If I remember correctly we would also need to fix the backend since it was serving cached response, and this hack was bypassing that.

Comment (jsnajdr) [reply packages/happychat-connection/src/use-happychat-auth.ts:47]: It's unlikely that the `key = Date.now()` hack would have any influence on backend. The query function doesn't use the `key` at all, it's not part of the REST request params.

**Pull request merged**
